compiler/luci/pass/src/FuseRsqrtPass.cpp

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright (c) 2024 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "luci/Pass/FuseRsqrtPass.h"
+#include "helpers/NodeFiller.h"
+
+#include <luci/IR/CircleNodes.h>
+#include <luci/Profile/CircleNodeOrigin.h>
+
+#include <cmath>
+#include <cassert>
+
+namespace
+{
+
+// Float comparison
+bool same(float a, float b) { return fabs(a - b) < 1e-5; }
+
+class RsqrtPattern
+{
+public:
+  RsqrtPattern(luci::CircleDiv *candidate)
+  {
+    assert(candidate); // FIX_CALLER_UNLESS
+    _div = candidate;
+  }
+
+#define CHECK_OR_FALSE(condition) \
+  if (not(condition))             \
+    return false;
+
+public:
+  bool matched()
+  {
+    // Check pattern
+    CHECK_OR_FALSE(luci::fill(&_div_const, &_sqrt).with_args_of(_div));
+    _ifm = loco::must_cast<luci::CircleNode *>(_sqrt->x());
+
+    CHECK_OR_FALSE(_div->fusedActivationFunction() == luci::FusedActFunc::NONE);
+
+    // Check div_const = 1
+    switch (_div->dtype())
+    {
+      case loco::DataType::S16:
+        CHECK_OR_FALSE(_div_const->quantparam() != nullptr);
+        CHECK_OR_FALSE(_div_const->quantparam()->scale.size() == 1);
+        CHECK_OR_FALSE(_div_const->quantparam()->zerop.size() == 1);
+        CHECK_OR_FALSE(_div_const->quantparam()->zerop.at(0) == 0);
+        CHECK_OR_FALSE(_div_const->size<loco::DataType::S16>() == 1);
+        CHECK_OR_FALSE(same(
+          1.0,
+          _div_const->at<loco::DataType::S16>(0) * _div_const->quantparam()->scale.at(0) == 1));
```

Comment:
there's `== 1` at the end, which makes this a bool...
is this intended?

Reply:
No. It works as I expected by coincidence.. I've updated the code. Thanks for finding!